H2MLauncher.Core/ViewModels/ServerBrowserViewModel.cs

```diff
@@ -74,20 +80,33 @@ public ServerBrowserViewModel(
             CheckUpdateStatusCommand = new AsyncRelayCommand(CheckUpdateStatusAsync);
             CopyToClipBoardCommand = new RelayCommand<ServerViewModel>(DoCopyToClipBoardCommand);
             SaveServersCommand = new AsyncRelayCommand(SaveServersAsync);
+            OpenUpdatePageInBrowserCommand = new AsyncRelayCommand(DoOpenUpdatePageInBrowserCommand, () => _updateStatus != "" && _updateStatus != "Restart for new version!");
             _logger = logger;
             _saveFileService = saveFileService;
             _errorHandlingService = errorHandlingService;
-            OpenUpdatePageInBrowserCommand = new RelayCommand(DoOpenUpdatePageInBrowserCommand, () => _updateStatus != "");
         }
 
-        private void DoOpenUpdatePageInBrowserCommand()
+        private async Task DoOpenUpdatePageInBrowserCommand()
         {
-            string destinationurl = "https://github.com/Bowhza/H2M-Launcher/releases/latest";
-            ProcessStartInfo sInfo = new(destinationurl)
+            ProgressBarVisibility = "Visible";
+            await _h2MLauncherService.UpdateLauncherToLatestVersion((double progress) =>
             {
-                UseShellExecute = true,
-            };
-            Process.Start(sInfo);
+                UpdateDownloadProgress = progress;
+                if (progress == 100)
+                {
+                    ProgressBarVisibility = "Hidden";
```

Comment:
Why not use result of method instead of checking progress? What happens if progress never is exactly 100?

Reply:
Well, then the download isn't finished. We want to show progress to the user, so we can't just wait on it to complete.

Reply:
i mean > 100 because it never hit exactly 100